### ISSUE TITLE
Ruby 1.8.7 support dropped?

### ISSUE THREAD

**Issue by juskoljo** [ohler55/oj#370]
Hi,

I updated oj from version 2.18.3 to 3.0.1 and got following errors while compiling:

```
$ gem install oj -v=3.0.1
Fetching: oj-3.0.1.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing oj:
  ERROR: Failed to build gem native extension.

        /Users/jussi/.rvm/rubies/ruby-1.8.7-p371/bin/ruby extconf.rb
>>>>> Creating Makefile for ruby version 1.8.7 on i686-darwin12.2.1 <<<<<
creating Makefile

make
gcc-4.2 -I. -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE   -DDATETIME_1_8=1 -DNEEDS_RATIONAL=1 -DRUBY_VERSION_MINOR=8 -DRUBY_VERSION_MAJOR=1 -DHAS_METHOD_ARITY=1 -DUSE_PTHREAD_MUTEX=1 -DHAS_DATA_OBJECT_WRAP=0 -DRUBY_VERSION=1.8.7 -DRUBY_TYPE=ruby -DHAS_STRUCT_MEMBERS=1 -DNO_TIME_ROUND_PAD=0 -DHAS_NANO_TIME=0 -DHAS_TOP_LEVEL_ST_H=1 -DHAS_ENCODING_SUPPORT=0 -DUSE_RB_MUTEX=0 -DIS_WINDOWS=0 -DHAS_EXCEPTION_MAGIC=1 -DHAS_IVAR_HELPERS=0 -DRUBY_VERSION_MICRO=7 -DHAS_PROC_WITH_BLOCK=0 -DHAS_RB_TIME_TIMESPEC=0 -DUNIFY_FIXNUM_AND_BIGNUM=0 -DRUBY_RUBY -Wall -fno-common -g -O2  -fno-common -pipe -fno-common   -c cache8.c
gcc-4.2 -I. -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE   -DDATETIME_1_8=1 -DNEEDS_RATIONAL=1 -DRUBY_VERSION_MINOR=8 -DRUBY_VERSION_MAJOR=1 -DHAS_METHOD_ARITY=1 -DUSE_PTHREAD_MUTEX=1 -DHAS_DATA_OBJECT_WRAP=0 -DRUBY_VERSION=1.8.7 -DRUBY_TYPE=ruby -DHAS_STRUCT_MEMBERS=1 -DNO_TIME_ROUND_PAD=0 -DHAS_NANO_TIME=0 -DHAS_TOP_LEVEL_ST_H=1 -DHAS_ENCODING_SUPPORT=0 -DUSE_RB_MUTEX=0 -DIS_WINDOWS=0 -DHAS_EXCEPTION_MAGIC=1 -DHAS_IVAR_HELPERS=0 -DRUBY_VERSION_MICRO=7 -DHAS_PROC_WITH_BLOCK=0 -DHAS_RB_TIME_TIMESPEC=0 -DUNIFY_FIXNUM_AND_BIGNUM=0 -DRUBY_RUBY -Wall -fno-common -g -O2  -fno-common -pipe -fno-common   -c circarray.c
gcc-4.2 -I. -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE   -DDATETIME_1_8=1 -DNEEDS_RATIONAL=1 -DRUBY_VERSION_MINOR=8 -DRUBY_VERSION_MAJOR=1 -DHAS_METHOD_ARITY=1 -DUSE_PTHREAD_MUTEX=1 -DHAS_DATA_OBJECT_WRAP=0 -DRUBY_VERSION=1.8.7 -DRUBY_TYPE=ruby -DHAS_STRUCT_MEMBERS=1 -DNO_TIME_ROUND_PAD=0 -DHAS_NANO_TIME=0 -DHAS_TOP_LEVEL_ST_H=1 -DHAS_ENCODING_SUPPORT=0 -DUSE_RB_MUTEX=0 -DIS_WINDOWS=0 -DHAS_EXCEPTION_MAGIC=1 -DHAS_IVAR_HELPERS=0 -DRUBY_VERSION_MICRO=7 -DHAS_PROC_WITH_BLOCK=0 -DHAS_RB_TIME_TIMESPEC=0 -DUNIFY_FIXNUM_AND_BIGNUM=0 -DRUBY_RUBY -Wall -fno-common -g -O2  -fno-common -pipe -fno-common   -c code.c
gcc-4.2 -I. -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE   -DDATETIME_1_8=1 -DNEEDS_RATIONAL=1 -DRUBY_VERSION_MINOR=8 -DRUBY_VERSION_MAJOR=1 -DHAS_METHOD_ARITY=1 -DUSE_PTHREAD_MUTEX=1 -DHAS_DATA_OBJECT_WRAP=0 -DRUBY_VERSION=1.8.7 -DRUBY_TYPE=ruby -DHAS_STRUCT_MEMBERS=1 -DNO_TIME_ROUND_PAD=0 -DHAS_NANO_TIME=0 -DHAS_TOP_LEVEL_ST_H=1 -DHAS_ENCODING_SUPPORT=0 -DUSE_RB_MUTEX=0 -DIS_WINDOWS=0 -DHAS_EXCEPTION_MAGIC=1 -DHAS_IVAR_HELPERS=0 -DRUBY_VERSION_MICRO=7 -DHAS_PROC_WITH_BLOCK=0 -DHAS_RB_TIME_TIMESPEC=0 -DUNIFY_FIXNUM_AND_BIGNUM=0 -DRUBY_RUBY -Wall -fno-common -g -O2  -fno-common -pipe -fno-common   -c compat.c
gcc-4.2 -I. -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I/Users/jussi/.rvm/rubies/ruby-1.8.7-p371/lib/ruby/1.8/i686-darwin12.2.1 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE   -DDATETIME_1_8=1 -DNEEDS_RATIONAL=1 -DRUBY_VERSION_MINOR=8 -DRUBY_VERSION_MAJOR=1 -DHAS_METHOD_ARITY=1 -DUSE_PTHREAD_MUTEX=1 -DHAS_DATA_OBJECT_WRAP=0 -DRUBY_VERSION=1.8.7 -DRUBY_TYPE=ruby -DHAS_STRUCT_MEMBERS=1 -DNO_TIME_ROUND_PAD=0 -DHAS_NANO_TIME=0 -DHAS_TOP_LEVEL_ST_H=1 -DHAS_ENCODING_SUPPORT=0 -DUSE_RB_MUTEX=0 -DIS_WINDOWS=0 -DHAS_EXCEPTION_MAGIC=1 -DHAS_IVAR_HELPERS=0 -DRUBY_VERSION_MICRO=7 -DHAS_PROC_WITH_BLOCK=0 -DHAS_RB_TIME_TIMESPEC=0 -DUNIFY_FIXNUM_AND_BIGNUM=0 -DRUBY_RUBY -Wall -fno-common -g -O2  -fno-common -pipe -fno-common   -c custom.c
custom.c:76:12: warning: implicit declaration of function 'rb_complex_new' is invalid in C99 [-Wimplicit-function-declaration]
    return rb_complex_new(rb_hash_aref(args, rb_id2str(real_id)), rb_hash_aref(args, rb_id2str(imag_id)));
           ^
custom.c:76:46: warning: implicit declaration of function 'rb_id2str' is invalid in C99 [-Wimplicit-function-declaration]
    return rb_complex_new(rb_hash_aref(args, rb_id2str(real_id)), rb_hash_aref(args, rb_id2str(imag_id)));
                                             ^
custom.c:186:12: warning: implicit declaration of function 'rb_rational_new' is invalid in C99 [-Wimplicit-function-declaration]
    return rb_rational_new(rb_hash_aref(args, rb_id2str(numerator_id)),
           ^
custom.c:373:15: error: use of undeclared identifier 'rb_eEncodingError'
            rb_raise(rb_eEncodingError, "Invalid type for raw JSON.\n");
                     ^
custom.c:464:10: warning: implicit declaration of function 'rb_obj_method_arity' is invalid in C99 [-Wimplicit-function-declaration]
        arity = rb_obj_method_arity(obj, oj_as_json_id);
                ^
custom.c:590:16: warning: implicit declaration of function 'rb_ivar_count' is invalid in C99 [-Wimplicit-function-declaration]
    cnt = (int)rb_ivar_count(obj);
               ^
custom.c:603:5: warning: implicit declaration of function 'rb_ivar_foreach' is invalid in C99 [-Wimplicit-function-declaration]
    rb_ivar_foreach(obj, dump_attr_cb, (VALUE)out);
    ^
custom.c:774:10: warning: implicit declaration of function 'RSTRUCT_GET' is invalid in C99 [-Wimplicit-function-declaration]
            v = RSTRUCT_GET(obj, i);
                ^
custom.c:860:29: error: use of undeclared identifier 'RUBY_T_FIXNUM'
    if (0 < type && type <= RUBY_T_FIXNUM) {
                            ^
custom.c:982:17: warning: implicit declaration of function 'rb_time_nano_new' is invalid in C99 [-Wimplicit-function-declaration]
                parent->val = rb_time_nano_new(ni->i, (long)nsec);
                              ^
8 warnings and 2 errors generated.
make: *** [custom.o] Error 1
```

With best regards,
Jussi

**Comment by juskoljo** [comment]
Didn't notice compatibility list https://github.com/ohler55/oj/blob/master/pages/Compatibility.md

**Comment by ohler55** [comment]
Oj 3.0 no longer supports Ruby 1.8. The 2.x versions must be used instead. If I missed that in the documentation updates let me know and I will correct it. The Compatibility page no longer lists 1.8.7.

**Comment by ohler55** [comment]
I was too slow in the reply.

**Comment by juskoljo** [comment]
Haha, no worries, my mistake & quick reaction 😄 

One of our systems is still running with Ruby 1.8.7, so I just fix the oj version to '< 3.0' 👍 

Thanks,
Jussi
